### DISCUSSION TITLE
Feeding image into neural network

### DISCUSSION THREAD

**Discussion by thestonechat** [BrainJS/brain.js#686] [Q&A]
Hey everyone!
I'm trying to make simple image recognition software for my ai assistant. I've tried using base64 data and binary but thats just too slow and TOTALLY doesnt work.
I've read somewhere that theres a way to feed in actual image itself. and im wondering how.
also please dont post some other discussion links because theres 90% chance I've seen it and it didnt help.
thanks for reading thru!

**Comment by mubaidr** [comment]
A lot of discussion about this topic here: https://github.com/BrainJS/brain.js/issues/352

In short: Convolutional neural network are in development.

**Reply by joschikbrunow** [reply]
Do you know, wether the development is finished or when it'll be finished?

**Reply by JeffMII** [reply]
Convolutional Neural Network would be a great addition to Brain.js. I'd like to know the progress of the development as well. Is there any ETA?

**Comment by robertleeplummerjr** [comment]
I fixed a wide variety of issue recently, and am now back at working on convolutions.  The missing layer that isn't yet built (of course the whole thing together isn't as well) is Pool layer, IIRC.
